src/utils/formHelperFunctions.js

```diff
@@ -330,14 +335,22 @@ function getFieldError(value, servar) {
 /**
  * Returns the list of DOM nodes corresponding to the field with the provided key.
  */
-function getElementsFromKey({ formRef, fieldKey }) {
+function setFormElementError({
+    formRef,
+    fieldKey,
+    message,
+    index = null,
+    servarType = ''
+}) {
+    if (servarType === 'pin_input') fieldKey = `${fieldKey}-0`;
     const singleOrList = formRef.current.elements[fieldKey];
-    const elements =
+    let elements =
         singleOrList instanceof RadioNodeList
             ? Array.from(singleOrList)
             : [singleOrList];
-
-    return elements.filter((e) => e);
+    elements = elements.filter((e) => e);
+    if (index) elements = [elements[index]];
```

Comment:
You should explicitly check for `index !== null` since this will fail if `index` is 0.